### PR TITLE
More updates for the docs

### PR DESCRIPTION
**Pull request by jsquyres**
Mostly having to do with DSO-vs-static issues, and mostly related to packagers.

See the individual commit messages for details.

### CODE REVIEW

File: docs/installing-open-mpi/packagers.rst [hunk 1]
```diff
@@ -80,15 +80,59 @@ running Open MPI's ``configure`` script.
...
+installed on a network filesystem (vs. being installed on a local
+filesystem).
+
+Specifically, when running thousands or MPI processes (or more) that
+each query and open dozens of DSO files on a network filesystem during
+:ref:``MPI_Init(3) <mpi_init>``, this can lead to a bottleneck of
+network performance.
+
+In this scenario, it is better to compile all the components
+statically as part of the core libraries such that this kind of
```
Comment (lrbison) [inline docs/installing-open-mpi/packagers.rst:130]: I would be pedantic here and replace "statically" with something that doesn't conjure .a files.  It's unfortunate the overlap in terms between DSO and SO make this verbiage hard.

**Review by lrbison — APPROVED** [review]

File: docs/installing-open-mpi/packagers.rst (hunk 1 above)
Comment (jsquyres) [reply docs/installing-open-mpi/packagers.rst]: Talked with @lrbison out-of-band and agreed in some revamped text for this section.  See new push.

**Review by lrbison — APPROVED** [review]
Thank you for writing all this up.

File: docs/installing-open-mpi/packagers.rst [hunk 2]
```diff
@@ -135,19 +198,121 @@ using ``--enable-mca-dso`` to selectively build some components as
...
+If you are building a package that includes support for one or more
+accelerators, it may be desirable to build accelerator-related
+components as DSOs (see the :ref:`static or DSO?
+<label-install-packagers-dso-or-not>` section for details).
+
+.. admonition:: Rationale
+   :class: tip
+
+   Accelerator hardware is expensive, and may only be present on
+   compute nodes in an HPC cluster.  Specifically: there may not be
```
Comment (edgargabriel) [inline docs/installing-open-mpi/packagers.rst:288]: I would suggest to change that to: "... and may only be present on some compute nodes in an HPC cluster."

**Review by edgargabriel — APPROVED** [review]

**Comment by jsquyres** [comment]
Did a new force-push to include @edgargabriel's 1-word suggested change.

**Pull request merged**